### PR TITLE
Add Action.ChooseFile for adaptive cards

### LINKED ISSUE #512: [Widgets] SSH keychain widget requires typing in a path
### Dev Home version

_No response_

### Windows build number

_No response_

### Other software

_No response_

### Steps to reproduce the bug

Description
this should be a file picker button. if the widget platform doesn't support it, we should push for it / create a host extension

### Expected result

_No response_

### Actual result

_No response_

### Included System Information

_No response_

### Included Extensions Information

_No response_

### PR DESCRIPTION
**Pull request by joadoumie**
## Summary of the pull request

The Action.ChooseFile adaptive card element allows a card author to launch a File Picker from Dev Home and obtain the chosen file. If no file is chosen, the "FilePath" field will be null. The author may use the action as an "inlineAction" on a Text input. The SSH Keychain widget uses the action this way.
image

This action is experimental while it goes through API review.

## PR checklist
- [ ] Closes #512
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

**Comment by joadoumie** [comment]
@krschau - this is the starting point we can work from, still got some refactoring, UI clean up etc. to finish!

File: tools/Dashboard/DevHome.Dashboard/ViewModels/WidgetViewModel.cs [hunk 1]
```diff
@@ -114,63 +114,65 @@ private async Task RenderWidgetFrameworkElementAsync()
     {
         await Task.Run(async () =>
         {
-            var cardTemplate = await Widget.GetCardTemplateAsync();
-            var cardData = await Widget.GetCardDataAsync();
+        var cardTemplate = await Widget.GetCardTemplateAsync();
```
Comment (krschau) [inline tools/Dashboard/DevHome.Dashboard/ViewModels/WidgetViewModel.cs:117]: Please fix/revert the indentation just so looking at the diff is easier. (control+KD in VS should do it)

Comment (joadoumie) [reply tools/Dashboard/DevHome.Dashboard/ViewModels/WidgetViewModel.cs:117]: rookie mistake - fixing now

File: common/Renderers/FilePickerAction.cs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+public class FilePickerParser : IAdaptiveActionParser
+{
+    public IAdaptiveActionElement FromJson(JsonObject inputJson, AdaptiveElementParserRegistration elementParsers, AdaptiveActionParserRegistration actionParsers, IList<AdaptiveWarning> warnings)
+    {
+        var filePickerAction = new FilePickerAction();
+
+        // Parse your custom properties here, e.g., ButtonText
+        filePickerAction.ButtonText = inputJson.GetNamedString("text", "Choose File");
+        filePickerAction.Verb = inputJson.GetNamedString("verb", string.Empty);
+
```
Comment (joadoumie) [inline common/Renderers/FilePickerAction.cs:129]: add title, tooltip + defaults

File: common/Renderers/FilePickerAction.cs [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    public string ButtonText { get; set; }
+
+    public string FilePath { get; set; } = string.Empty;
+
+    public string Verb { get; set; } = string.Empty;
+
+    public bool RoundedCorners { get; set; } = true;
+
+    public static readonly string CustomTypeString = "Action.FilePicker";
```
Comment (joadoumie) [inline common/Renderers/FilePickerAction.cs:43]: Change to Action.ChooseFile

**Review by guimafelipe — APPROVED** [review]

File: common/Renderers/ChooseFileAction.cs [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        var stringResource = new StringResource("DevHome.Common.pri", "DevHome.Common/Resources");
+
+        var chooseFileAction = new ChooseFileAction
+        {
+            Title = stringResource.GetLocalized("ChooseFileActionTitle"),
+            Tooltip = stringResource.GetLocalized("ChooseFileActionToolTip"),
+
+            // Parse the JSON properties of the action.
+            // The Verb ChooseFile is not meant to be localized.
+            Verb = inputJson.GetNamedString("verb", "ChooseFile"),
```
[the remark this answers is not in the document]
Comment (krschau) [reply common/Renderers/ChooseFileAction.cs:119]: This should get the verb specified in the card, and if one is not specified, use the default of "ChooseFile".

**Review by EricJohnson327 — APPROVED** [review]

**Pull request merged**